python/pyarrow/_flight.pyx

```diff
@@ -736,6 +743,12 @@ cdef class FlightEndpoint(_Weakrefable):
                     CLocation.Parse(tobytes(location)).Value(&c_location))
             self.endpoint.locations.push_back(c_location)
 
+        if expiration_time is not None:
+            self.endpoint.expiration_time = time_point(
+                microseconds(expiration_time.cast(timestamp("us")).value))
```

Comment:
Is microseconds correct?
I think that we should use nanoseconds: https://github.com/apache/arrow/blob/8068554a5d18129121f824978c7ee4a9c430640b/cpp/src/arrow/flight/types.h#L76-L88

Reply:
I'd prefer nanoseconds as well, but that did not work on macOS:
```
 /Users/runner/work/arrow/arrow/build/python/build/temp.macosx-10.9-universal2-cpython-311/_flight.cpp:31893:102: error: no matching conversion for functional-style cast from 'std::chrono::nanoseconds' (aka 'duration<long long, ratio<1LL, 1000000000LL>>') to 'std::chrono::system_clock::time_point' (aka 'time_point<std::chrono::system_clock>')
      __pyx_v_self->endpoint.expiration_time = ((std::optional<std::chrono::system_clock::time_point> )std::chrono::system_clock::time_point(std::chrono::nanoseconds(__pyx_t_12)));
                                                                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
https://github.com/EnricoMi/arrow/actions/runs/10083176212/job/27879168275#step:8:3720

Reply:
Can we use `std::chrono::duration_cast<Timestamp::duration>()` like https://github.com/apache/arrow/blob/1992cc64c74367579fb587f0431bf4340401c38a/cpp/src/arrow/flight/flight_internals_test.cc#L194-L195 with Cython?

Reply:
That did the trick, thanks for the pointer!

https://github.com/apache/arrow/blob/501d3a6d62d36e6d7c437d10c865535df44aeb33/python/pyarrow/_flight.pyx#L746-L748